civis/polling.py

```diff
@@ -173,8 +173,12 @@ def _set_api_result(self, result):
                     err_msg = str(result['error'])
                 except:  # NOQA
                     err_msg = str(result)
-                self._set_api_exception(exc=CivisJobFailure(err_msg, result),
-                                        result=result)
+                job_id = getattr(self, "job_id", None)
+                run_id = getattr(self, "run_id", None)
```

Comment:
`CivisFuture` subclasses `PollableResult`, but only a `CivisFuture` object has the `job_id` and `run_id` properties (which I moved from `ContainerFuture` "up" to `CivisFuture` a long time ago, #290). Though I don't know when realistically we'd have a `PollableResult` object without job/run ID, I'm happy with the `getattr` treatment here for now.